### ISSUE TITLE
wrong password?

### ISSUE THREAD

**Issue by rafaleo** [FluxionNetwork/fluxion#424]
-----
Hi. I used this for the first time. I'm talking about fake host method.

The behavior was as expected I think but the password was not working.
The user didn't know his password and typed whatever (random letters) (in android) to close the login page and it was closed at first attempt. The message at my end was about "successfully cracked" or so. The password doesn't match at my end - maybe I do something wrong? In instructions it's written the script redirects after correct password imputed. I don't know is it a bug or typical behavior as just not 100% cases can he handled. When I type the password in login page my attempts are saving to that file and nothing stops the host.

**Comment by MPX4132** [comment]
The script password verifier should work 100% of the time. If the script says the password was successfully cracked, it means you successfully acquired the correct secret passphrase and you should be able to connect to the target using that passphrase.

**Comment by rafaleo** [comment]
Thanks for response. I'm not able to connect with this password to this target.

**Comment by MPX4132** [comment]
If you can't connect using the password acquired, chances are the target client changed the password to the access point. If the script said you got the correct password, that must be it. You can try it on your own network if you have one, to make sure the script is working as intended because it seems something weird happened.

**Comment by rafaleo** [comment]
Ok. I was thinking about this possibility but is seemed very unlikely. Especially if the password was like random keyboarding and the time of all operations was in seconds. Anyway, I need to investigate and try again or as you said on some known network.

**Comment by MPX4132** [comment]
Yeah, because it's extremely unlikely the key is incorrect. You can check it yourself, if you want. Make a text file with the key in it:
```
SuperSecretKeyHere
```
Save it as key.txt to your desktop.

Copy the capture file for the target network to your desktop. The capture file can be found at:


`fluxion/attacks/Handshake Snooper/handshakes`

Now, run aircrack-ng to see if the key is the one that opens that handshake:
```
cd ~/Desktop
aircrack-ng -w key.txt XX:XX:XX:XX:XX:XX-something.cap
```

**Comment by rafaleo** [comment]
I've done the test but maybe got something wrong. Put the key in key.txt file alone (8 characters only)
```
 # aircrack-ng -w key.txt SAGEM_XXXX-XX:XX:XX:XX:XX:XX.cap

 SAGEM_XXXX-XX:XX:XX:XX:XX:XX.cap
Read 3 packets.

   #  BSSID              ESSID                     Encryption

   1  XX:XX:XX:XX:XX:XX  SAGEM_XXXX                WPA (1 handshake)

Choosing first network as target.

Opening SAGEM_XXXX-XX:XX:XX:XX:XX:XX.cap

                                 Aircrack-ng 1.2 rc4 r2858
Passphrase not in dictionary
                 1/0 keys tested 

      Time left: 0 seconds                                   inf%



Quitting aircrack-ng... 
```

**Comment by MPX4132** [comment]
I'm sorry, can you repeat the test again but change the last part to this:
```
cd ~/Desktop
aircrack-ng -a 2 -b XX:XX:XX:XX:XX:XX -w key.txt SSID_NAME_HERE-XX:XX:XX:XX:XX:XX.cap
```

Seems like I found a problem with the verifier, it's missing some parameters in the script but should probably still work because it has been working so far for the majority.

**Comment by deltaxflux** [comment]
Interesting...

**Comment by rafaleo** [comment]
I've got the same results. Without -b as well.

It's something wrong with aircrack at my end. When I input 2 passwords to key.txt I get:
- if there are 2 rows:
```
Passphrase not in dictionary
                 2/0 keys tested 

      Time left: 0 seconds                                   inf%
```
- if there are 3 rows so the last is empty:
```
Passphrase not in dictionary
                 2/1 keys tested 

      Time left: 0 seconds                                   200.00%
```

**Comment by MPX4132** [comment]
This is weird then, the key you got isn't the correct one but somehow passed through the verifier.

The problem is if we specify the attack parameter as WPA (`-a 2`) then WEP keys stop working. However, I don't think we need to specify the attack, since it's been working fine up to now, so I'm not sure what else could have caused the script to say you found the correct key if it wasn't the correct key.

**Comment by rafaleo** [comment]
So in this particular case it's most likely to be the fact of changing the password after attack. Maybe that person had some knowledge and ability to do it so quick (it had to be about 30s I guess as he realized earlier that gave a password into wrong host than I've got the message and connected). Maybe I'll try to interact again in near future and write if there's some solution for this (or not).

**Comment by MPX4132** [comment]
Well, the thing is if you used a capture file from before he changed the password, and that hash file (capture file) was unlocked by the password given to fluxion's captive portal, then that password should still be able to unlock the hash file (capture file). If you tested the password with the original hash file, something went wrong with fluxion and it shouldn't have said the correct password was given, however, if you got the handshake after the person changed the password, then the old password will not open the new handshake.

**Comment by rafaleo** [comment]
I don't have the original file from which the password was validated but have older one. The capture file I have and tested the password is from 11th of March, the password file from 13th and neither cap file, old and current one, don't resolve the passphrase. It could be a bug or the password was changed in that 2 days before my trial. It's unlikely in my guess. Maybe someone changes passwords every one or two days? It seems weird.

**Comment by rafaleo** [comment]
If it's some bug maybe it goes with the issue I got always a message "the handshake is corrupted" when I select it from disk - even if the script saved it few moments earlier. I always have to use freshly generated handshake for fluxion script. Others like aircrack or brutus work well.

**Comment by MPX4132** [comment]
If it's not the original then yeah, there's a very high chance the user changed the password to the router.

**Comment by MPX4132** [comment]
I don’t know that, I’m not familiar with the structure and layout of data in a handshake capture file, but I don’t believe that’s right. If you captured more data, then it’s possible to have the same handshake with for the same SSID and the same passphrase, you just captured more data.

**Comment by rad4day** [comment]
`does it mean if 2 handshake files for the same network obtained by the same method are different even by one byte (in content) then passwords are changed for sure?`
Of course not. That would defeat the purpose of a shared secret

**Comment by deltaxflux** [comment]
Sorry for the late answer. The answer is no, the exchange will be different every time. The keys protecting the handshake need be dynamic. Otherwise, there would be much worse problems.

**Comment by rafaleo** [comment]
I have some news to this issue. I don't believe everything is all right with this script or with my use of it. I've got another passphrase for this network using fluxion 0.23 rev. 44. This is what I got:
```
	FLUX 0.23 by deltax

	SSID: SAGEM_XXXX
	BSSID: xx:xx:xx:xx:xx:xx ()
	Channel: 6
	Security: WPA2
	Time: 04:32:15
	Password: lj..,b       @&#8361;(&#163; 
```
Unbelieveable password for me to use for a normal person. It looks like some encoding problems if they are not derived from encrypting errors. And are spaces allowed? Are non-unicode chars allowed? Or maybe they just were non-english letters? 

Later, I've check it with the original handshake and it didn't match:
```
> aircrack-ng -a 2 -b xx:xx:xx:xx:xx:xx -w /root/password.txt /root/handshakes/SAGEM_XXXX-xx:xx:xx:xx:xx:xx.cap
Opening /root/handshakes/SAGEM_XXXX-xx:xx:xx:xx:xx:xx.cap

                                 Aircrack-ng 1.2 rc4 r2858
Passphrase not in dictionary
                 0/7 keys tested 

      Time left: 0 seconds                                       0.00% 
```
It found 7 words, and I'm not surprised in this case because it looks worse than previous password (was like 'dtrfghyu'). I think it even could not be changed since there and I can't check it because handshakes are always "dynamically" different. 

Anyone have an idea what technically is wrong here?
